### PR TITLE
[API Compatibility No.188] Add target alias for F.nll_loss -part

### PR DESCRIPTION
**Pull request by Aidenwu0209**
### PR Category
User Experience

### PR Types
New features

### Description
- Add `@param_one_alias(["label", "target"])` for `paddle.nn.functional.nll_loss` to support PyTorch-style keyword argument `target` as alias of `label`.
- Update `nll_loss` docstring to mark `label` alias: `target`.
- Add legacy unit tests in `test/legacy_test/test_nll_loss.py` to verify `label=` and `target=` are equivalent for `reduction="none"` and `reduction="mean"`.
- Add an alias-equivalence case with `ignore_index=-100` and `reduction="sum"`.

No signature/default/numerical logic changes were introduced.

Related task: #76301 (No.188)

### 是否引起精度变化
否

### CODE REVIEW

**Review by zhwesky2010 — DISMISSED** [review]
LGTM

**Comment by Aidenwu0209** [comment]
Hi @zhwesky2010, I rebased this PR to keep only No.188 changes and updated the PR title/template fields per issue requirements. Could you please re-review when convenient? Thanks!

**Comment by Aidenwu0209** [comment]
@zhwesky2010 Hi, No.188 PR has been cleaned to keep only nll_loss changes and aligned with #76301 template requirements. Could you please re-review the latest commit when convenient? Thanks!

File: test/legacy_test/test_nll_loss.py [hunk 1]
```diff
@@ -1182,6 +1182,54 @@ def test_name(self):
                 self.assertTrue(res.name.startswith('nll_loss'))
 
 
+class TestNLLLossTargetAlias(unittest.TestCase):
+    def test_nll_loss_target_alias(self):
+        paddle.disable_static()
+        try:
```
Comment (zhwesky2010) [inline test/legacy_test/test_nll_loss.py]: 测试里面不需要用try，需要保证测试结果正确可运行

Comment (Aidenwu0209) [reply test/legacy_test/test_nll_loss.py]: 已按建议修改：去掉 try/finally，改为 `with base.dygraph.guard()` 管理动态图上下文，确保测试可运行且结构更简洁。请帮忙再看下，感谢！

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Comment by luotao1** [comment]
可以merge下develop，重新触发下CI

**Review by zhwesky2010 — APPROVED** [review]
